src/medVtkInria/medClutEditorToolBox/medClutEditorToolBox.cpp

```diff
@@ -148,6 +120,9 @@ medClutEditorToolBox::medClutEditorToolBox(QWidget *parent) : medToolBox(parent)
 
 medClutEditorToolBox::~medClutEditorToolBox(void)
 {
+    delete d->tables;
```

Comment:
As suggested by @fcollot, you should get rid of ```QList <medClutEditorTable *> * tables;``` as a pointer; just declare it as
```QList <medClutEditorTable *> tables;```

In ```void medClutEditorToolBox::updateSavedTables()```, add and replace:
```
void medClutEditorToolBox::updateSavedTables()
{
    foreach (medClutEditorTable* table, d->tables)
    {
        delete table;
    }
    d->tables.clear();
    d->tables = new QList<medClutEditorTable *>();  // <---- kill this line
```

Then, when instanciating the reader, change this line
```
    medXMLToLUTReader reader = medXMLToLUTReader(d->tables, this->getScene());
```
to 
```
    medXMLToLUTReader reader = medXMLToLUTReader(&d->tables, this->getScene());
```
This is ok since the reader goes out of scope at the end of the method.

Similarily, change all loops on the table from:
```
foreach (medClutEditorTable * table,*d->tables )
```
to
```
foreach (medClutEditorTable* table, d->tables )
```
Doing it this way you won't have to change anything in ```medXMLToLUTReader```

Reply:
Done, thank you @mjuhoor :)

![thankyou](https://user-images.githubusercontent.com/3910352/61120592-99e12280-a49d-11e9-94c8-538155714a40.gif)